lib/stellar/client.rb

```diff
@@ -192,165 +207,5 @@ def change_trust(
       horizon.transactions._post(tx: envelope_base64)
     end
     
-    Contract(C::KeywordArgs[
-      server: Stellar::KeyPair,
-      client: Stellar::KeyPair,
-      anchor_name: String,
-      timeout: C::Optional[Integer]
-    ] => String)
-    #
-    # Helper method to create a valid {SEP0010}[https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md]
-    # challenge transaction which you can use for Stellar Web Authentication.
-    #    
-    # @param server [Stellar::KeyPair] Keypair for server's signing account.
-    # @param client [Stellar::KeyPair] Keypair for the account whishing to authenticate with the server.
-    # @param anchor_name [String] Anchor's name to be used in the manage_data key.
-    # @param timeout [Integer] Challenge duration (default to 5 minutes).
-    #
-    # @return [String] A base64 encoded string of the raw TransactionEnvelope xdr struct for the transaction.
-    #
-    # = Example
-    # 
-    #   client = Stellar::Client.default_testnet
-    #   client.build_challenge_tx(server: server, client: user, anchor_name: anchor, timeout: timeout) 
-    # 
-    # @see {SEP0010: Stellar Web Authentication}[https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md]
-    def build_challenge_tx(server:, client:, anchor_name:, timeout: 300)
```

Comment:
I'm 💯 for moving the SEP10 functions out of Client and into a SEP10 class (in a backwards compatible way), I think that makes a ton of sense. Doing that in this PR means this PR is both refactoring and adding new functionality which [we try to avoid](https://github.com/stellar/go/blame/14a482c1286875cee8caef5ec1fa2fdb27e923c3/.github/pull_request_template.md#L11-L12). Would it be possible to do the refactor in another PR (should be fast to 👍), merge it master, then merge master into this PR?

Reply:
Sure, should be a fun git exercise.

Reply:
Actually, I'm a bit confused here. In order to refactor the methods in the Client class, the functions need to exist in SEP10 class, which doesn't exist in master yet. I think we need to merge this PR before refactoring.

Reply:
>In order to refactor the methods in the Client class, the functions need to exist in SEP10 class

Yup I'd expect this too. The refactor is to move the existing functions into a new SEP10 class. Then this PR just operates on adding new things to the SEP10 class.

However, if that's too much trouble at this point I think we can just move ahead.

Reply:
Lets just move ahead, I'll add the changes to Client back